File: security.php
<?php
session_start();

$max_requests = 10;
$time_frame = 60;
$block_time = 300;

$ip = $_SERVER['REMOTE_ADDR'];
$time = time();
$log_file = 'security_logs.txt';
$blocked_ips = 'blocked_ips.txt';

$blocked = file_exists($blocked_ips) ? file($blocked_ips, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES) : [];
if (in_array($ip, $blocked)) {
    die("403 Forbidden");
}

if (!isset($_SESSION['requests'])) {
    $_SESSION['requests'] = [];
}
array_push($_SESSION['requests'], $time);

$_SESSION['requests'] = array_filter($_SESSION['requests'], function ($t) use ($time, $time_frame) {
    return ($t > $time - $time_frame);
});

if (count($_SESSION['requests']) > $max_requests) {
    file_put_contents($blocked_ips, "$ip\n", FILE_APPEND);
    die("403 Forbidden - Too Many Requests");
}

if (!isset($_SERVER['HTTP_USER_AGENT']) || preg_match('/(bot|crawl|spider)/i', $_SERVER['HTTP_USER_AGENT'])) {
    file_put_contents($blocked_ips, "$ip\n", FILE_APPEND);
    die("403 Forbidden - Bot Detected");
}

$log_entry = "[$time] IP: $ip | User-Agent: " . $_SERVER['HTTP_USER_AGENT'] . "\n";
file_put_contents($log_file, $log_entry, FILE_APPEND);

?>

File: security.py
from flask import Flask, request, jsonify
import time
from collections import defaultdict
import re

app = Flask(__name__)

request_log = defaultdict(list)
blacklist = set()
RATE_LIMIT = 10
BLOCK_TIME = 300

bot_patterns = [
    re.compile(r"bot", re.IGNORECASE),
    re.compile(r"crawler", re.IGNORECASE),
    re.compile(r"spider", re.IGNORECASE),
]

@app.before_request
def block_attackers():
    ip = request.remote_addr
    user_agent = request.headers.get('User-Agent', '')
    current_time = time.time()
    
    if ip in blacklist:
        return jsonify({"error": "Access denied"}), 403
    
    if any(pattern.search(user_agent) for pattern in bot_patterns):
        blacklist.add(ip)
        return jsonify({"error": "Bot access denied"}), 403
    
    request_log[ip] = [t for t in request_log[ip] if current_time - t < 10]
    request_log[ip].append(current_time)
    
    if len(request_log[ip]) > RATE_LIMIT:
        blacklist.add(ip)
        return jsonify({"error": "Too many requests. You are temporarily blocked."}), 429

@app.route('/')
def home():
    return "Welcome to Secure Server!"

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)